### ISSUE TITLE
✨ CLI flag for running engine without cache volumes

### ISSUE THREAD

**Issue by levlaz** [dagger/dagger#8057]
### What are you trying to do?

Sometimes your build can get into a weird state, and its often related to cache volumes,. However, its hard to troubleshoot, it would be great to have a CLI option to run the engine while ignoring all cache volumes. 

### Why is this important to you?

This is a painful DX, if I had a flag that allowed me to enable or disable mounting cache volumes it would be a lot easier to rule out cache volumes as the issue when things go wrong. 

### How are you currently working around this?

Either commenting out the code that mounts cache volumes, or completely wiping out all volumes from my engine.

```
docker stop $(docker ps -a -q -f 'name=dagger-engine')
docker rm $(docker ps -a -q -f 'name=dagger-engine')
docker system prune -f -a --volumes
```

**Comment by jedevc** [comment]
x-linking to https://github.com/dagger/dagger/issues/6260, seems like some of this would be solved by that?

**Comment by levlaz** [comment]
> x-linking to #6260, seems like some of this would be solved by that?

Yeah 100% - being able to inspect cache would be awesome! I think having a one-off "build without cache" is still valuable though. Do you see that sort of thing being included in #6260

**Comment by sipsma** [comment]
I wouldn't be opposed to a one-off "disable cache" option meant to be used purely for debugging. I'd probably want it hidden away but agree it's nice while debugging to see if some problem is related to the cache without needing to necessarily throw it all away permanently.

Specifically, it would be useful to have this option for all cache, not just cache mounts, but something that only invalidates cache mounts specifically could make sense too.

---

Also related to https://github.com/dagger/dagger/issues/8004 in that part the goal there is to make the cache more observable + debuggable, which this would tie into.

**Comment by levlaz** [comment]
The reason why I specified cache mounts is because I think we can use the CACHEBUSTER pattern to clear the cache, but it does not seem to have any impact when you're using volumes. At least that was my experience recently when debugging some caching related issues that had to do with volumes.

Lev Lazinskiy
Head of Solutions @ dagger.io ( http://dagger.io )
e: ***@***.***

**Comment by markphelps** [comment]
Would also love this functionality! Thanks for posting the workaround @levlaz
